packages/jsii-pacmak/lib/targets/go/types/go-type.ts

```diff
@@ -60,7 +61,7 @@ export abstract class GoType {
       const instanceVar = this.proxyName[0];
       code.line(`${instanceVar} := ${this.proxyName}{}`);
       for (const base of bases) {
-        const baseEmbed = base.pkg === this.pkg ? base.proxyName : base.name;
+        const baseEmbed = embedForBase(this.pkg, this.proxyName, base).embed;
         code.line(
           `${JSII_RT_ALIAS}.InitJsiiProxy(&${instanceVar}.${baseEmbed})`,
```

Comment:
looks like we need to remove the `${instanceVar}.` prefix here?

Reply:
Possibly, checking...

Reply:
I believe it is required. 

Sample output:

```go
	_jsii_.RegisterClass(
		"jsii-calc.module2702.Class1",
		reflect.TypeOf((*Class1)(nil)).Elem(),
		[]_jsii_.Member{
			_jsii_.MemberMethod{JsiiMethod: "base", GoMethod: "Base"},
			_jsii_.MemberMethod{JsiiMethod: "typeName", GoMethod: "TypeName"},
		},
		func() interface{} {
			j := jsiiProxy_Class1{}
			_jsii_.InitJsiiProxy(&j.jsiiProxy_Class1_jcbBase)
			return &j
		},
	)
```

Reply:
You're correct this is required. I thought we were referencing the type here for some reason and not the embedded interface field.